build.cmd

```diff
@@ -1,47 +1,20 @@
 @echo off
 setlocal
 
-set _projectName=CommandLine
-set _solution=%_projectName%.sln
-set _codeProject=src\%_projectName%.csproj
-set _analyzerProject=analyzer\%_projectName%.Analyzer\%_projectName%.Analyzer.csproj
-set _testProject=test\%_projectName%.tests.csproj
-
 set _config=%1
 if not defined _config (
   set _config=Debug
 )
 
-echo Building Config '%_config%'
-echo Solution: '%_solution%'
-echo Code: '%_codeProject%'
-echo Analyzer: '%_analyzerProject%'
-echo Test: '%_testProject%'
-
-echo --------------------------
-echo !!! Restoring packages !!!
-echo --------------------------
-dotnet restore
-
 echo -------------------------
 echo !!! Cleaning solution !!!
 echo -------------------------
-dotnet clean %_solution%
+dotnet clean -c %_config%
 
 echo -------------------------
 echo !!! Building solution !!!
 echo -------------------------
-dotnet build %_solution% -c %_config%
-
-echo -------------------------
-echo !!! Building analyzer !!!
-echo -------------------------
-dotnet build %_analyzerProject% -c %_config%
-
-echo ---------------------
```

Comment:
For a developer workflow I think it would be nice if there still exists a batch file that will do everything for you:
- restore
- build
- test
- create package

I suggest to keep the current (small build.cmd) in a folder for appveyor and continue to have a build.cmd script that does everything.

That way we have both a good developer experience (single script to do everything) and we integrate with Appveyor.!

Reply:
In current setup, nuget package is generated automatically during build,  restore is covered in build process (unless a very old version of msbuild is being used), regarding tests, CI will catch failures and ppl can run tests in VS during development as well,  I feel the goal of this batch file can be replaced by CI

Reply:
I tried the build.cmd on the developer prompt for VS 2017 and restore was not called. I don't believe msbuild is calling restore by default.

While CI is great to quickly get validation of the change, I would like to be able to run the tests locally as well (usually pretty fast to do so).

While VS is great, since I am running pre-release versions of VS things don't always work well. Being able to run the tests on the command line is great!

How about having 2 scripts? One called `build.cmd` -- the one you modified, and another one called `buildAndTest.cmd` which will do what the old script was doing? I think they can be chained together (the `buildandtest.cmd` one would call the `build.cmd` one.)

Reply:
sure.  i dont really have time this week to make these changes   it will be great if you or i can fix it later

Reply:
I will do that -- I'll pull in the changes and do that. 

Do you want to update the commits with a bit more information? I try (not always successfully :)) to add more info into the commits about why / what the changes are. If you don't have time I'll update them later (I think GIT will correctly attribute the changes to you even if I do that).